src/middlewares/checkArticleOwnership.js

```diff
@@ -0,0 +1,48 @@
+import models from '../api/models';
+import statusCode from '../helpers/constants/status.codes';
+
+const { Article, User } = models;
+/**
+ * containing all user's model controllers (signup, login)
+ *
+ * @export
+ * @class CheckUserOwnership
+ */
+class CheckUserOwnership {
+  /**
+   * this is a middleware which checks if the user is registered in to our database.
+   *
+   * @author Frank Mutabazi
+   * @static
+   * @param {object} req the request
+   * @param {object} res the response to be sent
+   * @param { object } next the next route controller to be called
+   * @memberof CheckUserOwnership
+   * @returns {Object} res
+   */
+  static async checkOwner(req, res, next) {
+    const { user: { email } } = req.user;
+    const { slug } = req.params;
+    const result = await User.findOne({
+      where: {
+        email
+      }
+    });
+
+    const response = await Article.findOne({
```

Comment:
I think there is a process missing here. If for e.g the article doesn't exist, it's gonna raise an exception. 
My opinion, you should check first if the article exists before checking if the user is it's owner

Reply:
I have a middleware for such job @nkpremices